FEXCore/Source/Interface/Core/JIT/JIT.cpp

```diff
@@ -875,6 +876,13 @@ CPUBackend::CompiledCode Arm64JITCore::CompileCode(uint64_t Entry, uint64_t Size
     for (size_t i = 0; i < DebugData->GuestOpcodes.size(); i++) {
       const auto& GuestOpcode = DebugData->GuestOpcodes[i];
       auto& RIPEntry = JITRIPEntries[i];
+#if defined(ASSERTIONS_ENABLED) && ASSERTIONS_ENABLED
```

Comment:
The checks here are trivial (particularly compared to the two memory stores right after), so there's no need to restrict them to assertion-builds.

Reply:
This seems to be a code pattern used throughout FEX, unsure why

Reply:
I don't see a reason either, but a bad pattern being widespread doesn't make a compelling reason to stick to it either way.

Reply:
Yeah I'll clean it up in general

Reply:
Besides, there's a certain irony in using `__builtin_assume` annotations only if the macro to enable expensive assertions is active.

Reply:
The pattern here of wrapping `LOGMAN_THROW_{A,AA}_FMT` usage in additional `#if defined(ASSERTIONS_ENABLED) && ASSERTIONS_ENABLED` wrappers only matters in non-trivial places.

Like in ThreadManager.cpp this happens because it ensures correctness by locking a mutex and walking an array. So it does both, but we don't want that mutex locking in release builds.